### PR TITLE
Add schema visitor.

### PR DESCRIPTION
**Pull request by liurenjie1024**
This is the second part of introducing kudo, for a more complete pr please refer to https://github.com/NVIDIA/spark-rapids-jni/pull/2532. 

In this pr we introduced a post order visitor for schema and host column vectors.

### CODE REVIEW

File: src/main/java/com/nvidia/spark/rapids/jni/schema/SchemaVisitor.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * The order of visiting will be:
+ * <ol>
+ *     <li> Visit primitive field a1 </li>
+ *     <li> Visit primitive field a2</li>
+ *     <li> Visit struct field A, with results from a1 and a2</li>
+ *     <li> Previsit list field B</li>
+ *     <li> Visit primitive field b1</li>
+ *     <li> Visit list field B with results from b1 and previsit result. </li>
+ *     <li> Visit primitive field c1</li>
+ *     <li> Visit with results from columns A, B, and C</li>
```
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/schema/SchemaVisitor.java:47]: This just says "Visit with ..." which begs the question of visit _what_?  Is this visitTopSchema?

File: src/main/java/com/nvidia/spark/rapids/jni/schema/HostColumnsVisitor.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Visit a struct column.
+     * @param col the struct column to visit
+     * @param children the results of visiting the children
+     * @return the result of visiting the struct column
+     */
+    T visitStruct(HostColumnVectorCore col, List<T> children);
+
+    /**
+     * Visit a list column before actually visiting its child.
```
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/schema/HostColumnsVisitor.java:63]: It's a bit odd and asymmetric that lists get a previsit before children but structs do not.  Curious why this is needed?

**Review by jlowe — APPROVED** [review]
Overall lgtm.  Minor documentation comment and a question on asymmetric previsits.

**Review by GaryShen2008 — APPROVED** [review]
Approved the comment fixing after Jason's approval.

**Pull request merged**